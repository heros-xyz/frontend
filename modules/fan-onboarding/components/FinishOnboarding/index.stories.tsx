import { Box } from "@chakra-ui/react";
import type { Meta, StoryObj } from "@storybook/react";
import FinishOnboarding from ".";

const meta: Meta<typeof FinishOnboarding> = {
  title: "Fan/On Boarding/FinishOnboarding",
  component: FinishOnboarding,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;
type Story = StoryObj<typeof FinishOnboarding>;

export const FinishOnboardingComponent: Story = {
  render: (args) => (
    <Box h="100vh" bg="secondary">
      <FinishOnboarding />
    </Box>
  ),
};
