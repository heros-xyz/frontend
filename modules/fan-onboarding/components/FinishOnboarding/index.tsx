import React from "react";
import { Box, Text } from "@chakra-ui/react";
import { useRouter } from "next/router";
import { ArrowRight } from "@/components/svg/ArrowRight";
import { FanOnboardingSuccess } from "@/components/svg/FanOnboardingSuccess";
import HerosOnboardingWrapperNew from "@/components/ui/HerosOnboardingWrapperNew";

const FinishOnboarding = () => {
  const router = useRouter();
  return (
    <HerosOnboardingWrapperNew
      Icon={
        <FanOnboardingSuccess
          w={{ base: "90px", xl: "144px" }}
          h={{ base: "90px", xl: "144px" }}
          color={{ base: "#FFFAE8", xl: "secondary" }}
        />
      }
      textButton="Start Discovering"
      IconButton={<ArrowRight />}
      onSubmit={async () => {
        await fetch("/api/set-first-login");
        router.push("/fan");
      }}
      bgIconColor="secondary"
      isSuccessPage
    >
      <Box textAlign={{ base: "center", lg: "left" }} mb={5} color="primary">
        <Text
          fontFamily="heading"
          fontSize={{ base: "2xl", lg: "5xl" }}
          fontWeight="bold"
          mb={3}
          textTransform="capitalize"
        >
          Account set up Successfully
        </Text>
        <Text fontSize={{ base: "xs", xl: "md" }} color="grey.300">
          Your account has been set up successfully.
        </Text>
      </Box>
    </HerosOnboardingWrapperNew>
  );
};

export default FinishOnboarding;
