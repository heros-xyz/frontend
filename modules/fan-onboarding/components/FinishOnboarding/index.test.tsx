import { render } from "@testing-library/react";
import mockRouter from "next-router-mock";
import FinishOnboarding from ".";

jest.mock("next/router", () => require("next-router-mock"));

it("renders homepage unchanged", () => {
  mockRouter.push("/fan");
  const { container } = render(<FinishOnboarding />);
  expect(container).toMatchSnapshot();
});
