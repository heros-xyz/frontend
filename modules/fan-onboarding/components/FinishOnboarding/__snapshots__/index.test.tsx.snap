// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="css-16tif5l"
  >
    <div
      class="chakra-container css-10klw3m"
    >
      <div
        class="css-12ij4x2"
      />
      <div
        class="css-8qr4sb"
      >
        <div
          class="css-tayfq6"
        >
          <div
            class="css-6ywt87"
          >
            <svg
              class="chakra-icon css-5xjrgh"
              focusable="false"
              viewBox="0 0 151 150"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="75.5"
                cy="75"
                fill="white"
                r="74.5"
                stroke="#DDDDDD"
              />
              <g
                clip-path="url(#clip0_1512_24870)"
              >
                <path
                  d="M113.122 117.593C115.212 116.965 120.02 115.919 120.02 113.512C120.125 104.198 118.662 72.1745 116.885 64.6396C115.108 56.7908 87.6163 26.0233 81.6581 30.4187C75.909 34.814 88.6616 53.3373 88.6616 53.3373C88.6616 53.3373 73.1912 40.9885 68.2783 46.535C63.2609 52.2908 78.1041 64.0117 78.6268 65.3722C78.6268 65.3722 63.2609 56.1629 61.0658 62.0233C59.6024 65.8954 62.1111 68.6164 71.8323 74.2675C77.7905 77.721 72.8776 110.791 72.9822 116.442C72.8776 121.047 102.46 121.047 113.122 117.593Z"
                  fill="#FAC484"
                />
                <path
                  d="M102.565 54.8025C96.3977 54.1746 83.6451 54.8025 81.0318 68.1979C77.8959 84.3141 92.321 83.5816 92.5301 75.7327C92.7391 66.8374 106.433 67.57 107.06 74.3723"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M87.7219 102.837C83.7497 99.4885 82.0773 95.9304 81.7637 90.9072C81.1365 82.4304 91.4849 79.5002 93.5755 85.8839C94.9344 89.9653 97.4431 93.2095 100.161 93.6281C100.161 93.6281 107.896 94.256 106.642 87.1397"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M39.9517 50.3023C40.6834 47.4768 40.2653 46.2209 41.6242 46.2209C42.774 46.2209 42.3559 47.7907 42.8786 49.4651C43.6103 52.0814 44.4465 55.1163 50.4047 55.8489C53.4361 56.2675 52.8089 57.7326 51.1364 58.0465C50.1956 58.2558 50.1956 58.1512 48.7322 58.3605C43.8193 59.3023 43.1921 63.593 42.6695 66.8372C42.3559 68.7209 40.997 68.7209 40.5789 66.2093C40.0563 63.4884 40.1608 58.6744 32.2165 58.2558C30.5441 58.1512 30.5441 56.7907 31.7984 56.4768C35.7705 55.5349 38.4883 56.1628 39.9517 50.3023Z"
                  fill="#FFCC00"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M50.0913 62.4419C50.5094 60.7675 50.3003 60.0349 51.032 60.0349C51.6592 60.0349 51.4502 60.9768 51.7638 62.0233C52.1819 63.5931 52.7045 65.3721 56.2585 65.7907C58.0355 66 57.7219 66.9419 56.6766 67.1512C56.154 67.2558 56.154 67.1512 55.2132 67.3605C52.2864 67.8838 51.9728 70.5 51.6592 72.3837C51.4502 73.5349 50.7185 73.5349 50.4049 72.0698C50.0913 70.5 50.1958 67.5698 45.492 67.3605C44.5512 67.3605 44.5512 66.5233 45.2829 66.314C47.5826 65.4768 49.1505 65.8954 50.0913 62.4419Z"
                  fill="#FFCC00"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M55.9444 51.0351C56.258 49.8839 56.0489 49.4653 56.5716 49.4653C56.9897 49.4653 56.8851 50.0932 57.0942 50.7211C57.4078 51.7677 57.7214 52.9188 60.021 53.2328C61.1709 53.4421 60.9618 53.9653 60.3346 54.07C59.9165 54.1746 60.021 54.07 59.3939 54.1746C57.5123 54.4886 57.3033 56.163 57.0942 57.4188C56.9897 58.1514 56.467 58.1514 56.258 57.2095C56.0489 56.163 56.0489 54.2793 53.0175 54.1746C52.3904 54.1746 52.3904 53.6514 52.913 53.4421C54.2719 53.1281 55.3172 53.3374 55.9444 51.0351Z"
                  fill="#FFCC00"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_1512_24870"
                >
                  <rect
                    fill="white"
                    height="90"
                    transform="translate(30.125 30)"
                    width="90"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <div
            class="css-yzqqny"
          >
            <div
              class="css-0"
            >
              <div
                class="css-w8qtk9"
              >
                <p
                  class="chakra-text css-1x6r89o"
                >
                  Account set up Successfully
                </p>
                <p
                  class="chakra-text css-5zhaam"
                >
                  Your account has been set up successfully.
                </p>
              </div>
              <div
                class="css-lrby83"
              >
                <button
                  class="chakra-button css-k6pd2s"
                  type="button"
                >
                  Start Discovering
                  <span
                    class="css-9n3bbc"
                  >
                    <svg
                      class="chakra-icon css-iz364c"
                      focusable="false"
                      viewBox="0 0 15 14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M1.25 7L13.75 7M13.75 7L8.125 1.375M13.75 7L8.125 12.625"
                        stroke="#D2FFFA"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="1.5"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
          <button
            class="chakra-button css-1mv50kv"
            type="button"
          >
            Start Discovering
            <span
              class="css-9n3bbc"
            >
              <svg
                class="chakra-icon css-iz364c"
                focusable="false"
                viewBox="0 0 15 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1.25 7L13.75 7M13.75 7L8.125 1.375M13.75 7L8.125 12.625"
                  stroke="#D2FFFA"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <p
        class="chakra-text css-1n4wwj3"
      />
    </div>
  </div>
</div>
`;
